From bc3fcc9462ef4ba3ae66593cbaf47bf7af703ed3 Mon Sep 17 00:00:00 2001
From: Namjae Jeon <namjae.jeon@samsung.com>
Date: Fri, 19 Mar 2021 13:51:15 +0900
Subject: [PATCH] cifsd: fix static checker warning from
 smb_direct_post_send_data()
Git-commit: bc3fcc9462ef4ba3ae66593cbaf47bf7af703ed3
Patch-mainline: v5.15-rc1
References: bsn#12
Modified-by-SEL: No


Dan reported static checker warning:
 fs/cifsd/transport_rdma.c:1168 smb_direct_post_send_data()
 warn: missing error code 'ret'

This patch add missing ret error code.

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/cifsd/transport_rdma.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/cifsd/transport_rdma.c b/fs/cifsd/transport_rdma.c
index d235051dc5b1..45b76847f1e7 100644
--- a/fs/cifsd/transport_rdma.c
+++ b/fs/cifsd/transport_rdma.c
@@ -1165,6 +1165,7 @@ static int smb_direct_post_send_data(struct smb_direct_transport *t,
 				sg, SMB_DIRECT_MAX_SEND_SGES-1, DMA_TO_DEVICE);
 		if (sg_cnt <= 0) {
 			ksmbd_err("failed to map buffer\n");
+			ret = -ENOMEM;
 			goto err;
 		} else if (sg_cnt + msg->num_sge > SMB_DIRECT_MAX_SEND_SGES-1) {
 			ksmbd_err("buffer not fitted into sges\n");
-- 
2.33.0

